### ISSUE TITLE
Failed to build for android after upgrading to TNS@4

### ISSUE THREAD

**Issue by anaszgh** [NativeScript/nativescript-cli#3587]
Platform: Android
OS: MacOS
Nativescript Version: 4.0.1

previously i was using NativeScript version 3.4.1 and it was working correctly, regarding some issues i was encouraged to upgrade to version 4, after doing so i keep getting the following build error 
on `tns build android`

```
Failed to execute aapt
com.android.ide.common.process.ProcessException: Failed to execute aapt
        at com.android.builder.core.AndroidBuilder.processResources(AndroidBuilder.java:796)
        at com.android.build.gradle.tasks.ProcessAndroidResources.invokeAaptForSplit(ProcessAndroidResources.java:551)
        at com.android.build.gradle.tasks.ProcessAndroidResources.doFullTaskAction(ProcessAndroidResources.java:285)
        at com.android.build.gradle.internal.tasks.IncrementalTask.taskAction(IncrementalTask.java:109)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)
        at org.gradle.api.internal.project.taskfactory.DefaultTaskClassInfoStore$IncrementalTaskAction.doExecute(DefaultTaskClassInfoStore.java:173)
        at org.gradle.api.internal.project.taskfactory.DefaultTaskClassInfoStore$StandardTaskAction.execute(DefaultTaskClassInfoStore.java:134)
        at org.gradle.api.internal.project.taskfactory.DefaultTaskClassInfoStore$StandardTaskAction.execute(DefaultTaskClassInfoStore.java:121)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:122)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:197)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:107)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:111)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:92)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:70)
        at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:63)
        at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutputCachingStateExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:58)
        at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:88)
        at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:52)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:52)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker$1.run(DefaultTaskGraphExecuter.java:248)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:197)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:107)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:241)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:230)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.processTask(DefaultTaskPlanExecutor.java:124)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.access$200(DefaultTaskPlanExecutor.java:80)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:105)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:99)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.execute(DefaultTaskExecutionPlan.java:625)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.executeWithTask(DefaultTaskExecutionPlan.java:580)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.run(DefaultTaskPlanExecutor.java:99)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
        at java.lang.Thread.run(Thread.java:748)
Caused by: java.util.concurrent.ExecutionException: java.util.concurrent.ExecutionException: com.android.tools.aapt2.Aapt2Exception: AAPT2 error: check logs for details
        at com.google.common.util.concurrent.AbstractFuture.getDoneValue(AbstractFuture.java:503)
        at com.google.common.util.concurrent.AbstractFuture.get(AbstractFuture.java:482)
        at com.google.common.util.concurrent.AbstractFuture$TrustedFuture.get(AbstractFuture.java:79)
        at com.android.builder.core.AndroidBuilder.processResources(AndroidBuilder.java:794)
        ... 48 more
Caused by: java.util.concurrent.ExecutionException: com.android.tools.aapt2.Aapt2Exception: AAPT2 error: check logs for details
        at com.google.common.util.concurrent.AbstractFuture.getDoneValue(AbstractFuture.java:503)
        at com.google.common.util.concurrent.AbstractFuture.get(AbstractFuture.java:462)
        at com.google.common.util.concurrent.AbstractFuture$TrustedFuture.get(AbstractFuture.java:79)
        at com.android.builder.internal.aapt.v2.QueueableAapt2.lambda$makeValidatedPackage$1(QueueableAapt2.java:179)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        ... 1 more
Caused by: com.android.tools.aapt2.Aapt2Exception: AAPT2 error: check logs for details
        at com.android.builder.png.AaptProcess$NotifierProcessOutput.handleOutput(AaptProcess.java:463)
        at com.android.builder.png.AaptProcess$NotifierProcessOutput.err(AaptProcess.java:415)
        at com.android.builder.png.AaptProcess$ProcessOutputFacade.err(AaptProcess.java:332)
        at com.android.utils.GrabProcessOutput$1.run(GrabProcessOutput.java:104)

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:processDebugResources'.
> Failed to execute aapt
```

**Please note that this error appears only on Android builds, IOS works fine**

**Comment by NickIliev** [comment]
@anaszgh 

### Which platform(s) does your issue occur on?
iOS/Android/Both

### Please provide the following version numbers that your issue occurs with:
- CLI: (run `tns --version` to fetch it)
- Cross-platform modules: (check the 'version' attribute in the
`node_modules/tns-core-modules/package.json` file in your project)
- Runtime(s): (look for the `"tns-android"` and `"tns-ios"` properties in the
`package.json` file of your project)
- Plugin(s): (look for the version number in the `package.json` file of your
project)

### Please tell us how to recreate the issue in as much detail as possible.

**Comment by anaszgh** [comment]
Platform: Android only
CLI: 4.0.1
Cross-Platform modules: 4.0.0
Runtime: Android:4.0.0

**Comment by anaszgh** [comment]
The problem is that the issue occurs only on thi project, after upgrading it from tns 3.4.1 to tns 4, Creating new project in the cli will build successfully.

**Comment by NickIliev** [comment]
@anaszgh check [this solution ](https://github.com/NativeScript/android-runtime/issues/947#issuecomment-365211415) and also consider posting the content of your `package.json`
```
Plugin(s): (look for the version number in the package.json file of your
project)
```

Make sure that your project is rebuilt entirely
```
rm -rf node_modules platforms hooks
npm i
tns run android
```

**Comment by anaszgh** [comment]
@NickIliev here is the content of package.json
```
{
	"description": "NativeScript Application",
	"license": "SEE LICENSE IN <your-license-filename>",
	"readme": "NativeScript Application",
	"repository": "<fill-your-repository-here>",
	"nativescript": {
		"id": "com.eventrest.mobile",
		"tns-android": {
			"version": "4.0.1"
		}
	},
	"dependencies": {
		"@angular/animations": "~5.2.0",
		"@angular/common": "~5.2.0",
		"@angular/compiler": "~5.2.0",
		"@angular/core": "~5.2.0",
		"@angular/forms": "~5.2.0",
		"@angular/http": "~5.2.0",
		"@angular/platform-browser": "~5.2.0",
		"@angular/platform-browser-dynamic": "~5.2.0",
		"@angular/router": "~5.2.0",
		"nativescript-angular": "~5.2.0",
		"nativescript-cardview": "^2.0.5",
		"nativescript-carousel": "^3.1.1",
		"nativescript-geolocation": "^4.2.6",
		"nativescript-grid-view": "^3.4.0",
		"nativescript-imagepicker": "^6.0.1",
		"nativescript-loading-indicator": "^2.4.0",
		"nativescript-push-notifications": "^1.1.3",
		"nativescript-secure-storage": "^2.2.2",
		"nativescript-snackbar": "^2.0.1",
		"nativescript-social-login": "^3.0.8",
		"nativescript-social-share": "^1.5.0",
		"nativescript-theme-core": "~1.0.4",
		"reflect-metadata": "~0.1.8",
		"rxjs": "~5.5.2",
		"tns-core-modules": "^4.0.0",
		"zone.js": "~0.8.2"
	},
	"devDependencies": {
		"babel-traverse": "6.26.0",
		"babel-types": "6.26.0",
		"babylon": "6.18.0",
		"lazy": "1.0.11",
		"nativescript-dev-typescript": "~0.6.0",
		"typescript": "~2.6.2"
	}
}
```
I'll try the proposed fix and let you know, Thanks.

**Comment by anaszgh** [comment]
Hello @NickIliev 

Were you able to reproduce the issue, I have uninstalled android-sdk and installed it again on different path (as advised on slack channel), then i ran `tns doctor` with no issues.

Then i have removed the Hooks, Platforms and node_modules folders as advised buy you.
But the issue remains the same when running `tns build android`.

**Comment by anaszgh** [comment]
@NickIliev i found the issue, and was able to reproduce it on a new application using 
`tns create {name} --ng`

if you Uncomment this section in the app.gradle
```
// Uncomment to add recyclerview-v7 dependency
dependencies {
	compile 'com.android.support:recyclerview-v7:+'
}
```

you'll face the issue, comment it again and the build succeeds, and please note that this happens only on NativeScript 4

**Comment by NickIliev** [comment]
@anaszgh I  was able to reproduce the issue with your package.json content and I've noticed that your version of some of the plugins is not updated t the latest. In specific the plugin nativescript-social-login has released a version 4.0.1 for compatibility with NativeScript 4.0.0. Updating all dependencies to latest has resolved the issue on my test project.

```
	"dependencies": {
		"@angular/animations": "~5.2.0",
		"@angular/common": "~5.2.0",
		"@angular/compiler": "~5.2.0",
		"@angular/core": "~5.2.0",
		"@angular/forms": "~5.2.0",
		"@angular/http": "~5.2.0",
		"@angular/platform-browser": "~5.2.0",
		"@angular/platform-browser-dynamic": "~5.2.0",
		"@angular/router": "~5.2.0",
		"nativescript-angular": "~5.3.0",
		"nativescript-cardview": "^3.0.1",
		"nativescript-carousel": "^3.1.1",
		"nativescript-geolocation": "^4.2.6",
		"nativescript-grid-view": "^3.4.0",
		"nativescript-imagepicker": "^6.0.1",
		"nativescript-loading-indicator": "^2.4.0",
		"nativescript-push-notifications": "^1.1.3",
		"nativescript-secure-storage": "^2.2.2",
		"nativescript-snackbar": "^3.0.0",
		"nativescript-social-login": "^4.0.1",
		"nativescript-social-share": "^1.5.0",
		"nativescript-theme-core": "~1.0.4",
		"reflect-metadata": "~0.1.8",
		"rxjs": "~5.5.2",
		"tns-core-modules": "^4.0.0",
		"zone.js": "~0.8.2"
	},
	"devDependencies": {
		"babel-traverse": "6.26.0",
		"babel-types": "6.26.0",
		"babylon": "6.18.0",
		"lazy": "1.0.11",
		"nativescript-dev-typescript": "~0.6.0",
		"typescript": "~2.6.2"
	}
```

You might also need to synchronize the version of `com.google.android.gms ` as the library is used by multiple plugins and they need to use a single version (otherwise the build will fail with conflict)

**Comment by anaszgh** [comment]
@NickIliev Thank you.
